pkg/group/service_test.go

```diff
@@ -0,0 +1,318 @@
+package group
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/dhis2-sre/im-user/internal/errdef"
+	"github.com/dhis2-sre/im-user/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+)
+
+func Test_service_GetClusterConfiguration_Happy(t *testing.T) {
+	id := uint(1)
+	groupName := "whatever"
+	clusterConfiguration := &model.ClusterConfiguration{
+		Model:                   gorm.Model{ID: id},
+		GroupName:               groupName,
+		KubernetesConfiguration: nil,
+	}
+
+	repository := &mockGroupRepository{}
+	repository.
+		On("getClusterConfiguration", groupName).
+		Return(clusterConfiguration, nil)
+	userService := &mockUserService{}
+
+	service := NewService(repository, userService)
+
+	cc, err := service.GetClusterConfiguration(groupName)
+	require.NoError(t, err)
```

Comment:
As `require.NoError(t, err)` is asserting on the method under test I would keep it together with the `assert` group at the end of the test. I like to arrange tests into AAA arrange, act, assert blocks if possible. So the method under test is easy to spot 😄 Some of your tests do roughly follow that pattern 😄